VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room.component.ts

```diff
@@ -0,0 +1,705 @@
+        this.logger.debug(`${this.componentLoggerPrefix} Judge responded to start hearing confirmation`, {
+            conference: this.conferenceId,
+            status: this.vhConference.status,
+            confirmStart: actionConfirmed
+        });
+        this.displayConfirmStartHearingPopup = false;
+        if (actionConfirmed) {
+            this.startHearing();
+        } else {
+            this.focusService.restoreFocus();
+        }
+    }
+
+    startHearing() {
+        this.audioRecordingService.restartActioned = false;
+        this.logger.debug(`${this.componentLoggerPrefix} Judge clicked start/resume hearing`, {
+            conference: this.conferenceId,
+            status: this.vhConference.status
+        });
+
+        this.store.dispatch(VideoCallHostActions.startHearing({ conferenceId: this.conferenceId }));
+    }
+
+    onJoinConfirmAnswered(actionConfirmed: boolean) {
+        this.logger.debug(`${this.componentLoggerPrefix} Judge responded to join hearing confirmation`, {
+            conference: this.conferenceId,
+            status: this.vhConference.status,
+            confirmStart: actionConfirmed
+        });
+        this.displayJoinHearingPopup = false;
+        if (actionConfirmed) {
+            this.joinHearingInSession();
+        }
+    }
+
+    joinHearingInSession() {
+        this.store.dispatch(VideoCallHostActions.joinHearing({ conferenceId: this.vhConference.id, participantId: this.vhParticipant.id }));
+    }
+
+    audioRestartCallback(continueWithNoRecording: boolean) {
+        this.continueWithNoRecording = continueWithNoRecording;
+        this.audioErrorRetryToast = null;
+    }
+
+    verifyAudioRecordingStream() {
+        /// If audio recording is required,
+        // has not been confirmed by user, to continue without recording,
+        // video is open,
+        // the alert isn't open already,
+        // Recording is not paused
+        // and the audio streaming agent cannot be validated, then show the alert
+        if (
+            this.vhConference.audioRecordingRequired &&
+            !this.continueWithNoRecording &&
+            this.showVideo &&
+            !this.audioErrorRetryToast &&
+            !this.recordingPaused &&
+            !this.audioRecordingService.wowzaAgent?.isAudioOnlyCall
+        ) {
+            this.logWowzaAlert();
+            this.showAudioRecordingRestartAlert();
+        }
+    }
+
+    private onShouldReload(): void {
+        window.location.reload();
+    }
+
+    private onShouldUnload(): void {
+        this.cleanUp();
+    }
+
+    private setUpSubscribers() {
+        this.subscribeToClock();
+        this.startEventHubSubscribers();
+        this.connectToPexip();
+        this.registerMediaStatusPublisher();
+        this.startVideoCallEventSubscribers();
+    }
+
+    private registerMediaStatusPublisher() {
+        this.userMediaService.isAudioOnly$.pipe(takeUntil(this.destroyedSubject)).subscribe(async audioOnly => {
```

Comment:
I'm not convinced this needs to live here. I was going to wait for your changes to see where it would go but it still doesn't fit right for me. What do you think? One for another day?

Reply:
yeah if we could consider this for the future ticket that would be great